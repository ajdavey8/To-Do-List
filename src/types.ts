export type CreateAction = {
  type: 'CREATE',
  payload: {
    name: string
    description: string
  }
}

export type UpdateAction = {
  type: 'UPDATE'
  payload: {
    id: number
    name: string
    description: string
  }
}

export type DeleteAction = {
  type: 'DELETE'
  payload: {
    id: number
  }
}

export type ResetAction = {
  type: 'RESET'
}

export type Actions = CreateAction | UpdateAction | DeleteAction | ResetAction

export interface Todo {
  id: number
  name: string
  description: string
  creationDate: string
}

export type NewTodo = Omit<Todo, 'creationDate' | 'id'>

export type UpdatedTodo = Omit<Todo, 'creationDate'> 