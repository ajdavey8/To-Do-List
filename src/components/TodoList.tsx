import React from 'react';
import styled from 'styled-components';

import { Todo } from '../types';

import TodoListItem from './TodoListItem';

const Container = styled.section`
  text-align: center;
  padding: 24px 16px;

  @media only screen and (min-width: 768px) {
    background: #ffffff;
    box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
    border-radius: 3px;
    max-width: 600px;
    padding: 24px;
    margin: 0 auto;
  }
`

interface TodoList {
  todos: Todo[]
  onUpdate: (action: any) => void
  onDelete: (id: number) => void
}

const TodoList = ({ todos, onUpdate, onDelete }: TodoList) => (
  <Container>
    {todos?.length ?
      (todos.map(todo => (
        <TodoListItem
          key={todo.id}
          id={todo.id}
          name={todo.name}
          description={todo.description}
          onUpdate={onUpdate}
          onDelete={onDelete}
        />
      )))
      : <p>No current todos</p>
    }
  </Container>
)

export default TodoList