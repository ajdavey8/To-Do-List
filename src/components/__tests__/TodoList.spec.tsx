import React from 'react';
import { render } from '@testing-library/react'
import TodoList from '../TodoList'
import { Todo } from '../../types';

const onUpdate = jest.fn()
const onDelete = jest.fn()

type Todos = Todo[]

const emptyTodos: Todos = []

const setup = (todos = emptyTodos) => {
  const utils = render(
    <TodoList
      todos={todos}
      onUpdate={onUpdate}
      onDelete={onDelete}
    />
  )

  return {
    ...utils
  }
}

const todos: Todo[] = [
  {
    id: 1,
    name: 'test',
    description: 'testing',
    creationDate: Date()
  }
]

describe('<TodoList>', () => {
  it('should return a message that the todo list is empty', () => {
    const { getByText } = setup()
    getByText('No current todos')
  })

  it('should return a todo', () => {
    const { asFragment } = setup(todos)
    expect(asFragment).toMatchSnapshot();
  })
})