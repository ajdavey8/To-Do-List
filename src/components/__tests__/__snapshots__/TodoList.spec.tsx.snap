// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList> should return a todo 1`] = `[Function]`;
